src/lms-ds-loader/edfi_lms_ds_loader/mssql_lms_operations.py

```diff
@@ -218,6 +218,74 @@ def insert_new_records_to_production_for_section(
   AND
     SourceSystem = stg.SourceSystem
 )
+"""
+
+        row_count = self._exec(statement)
+        logger.debug(f"Inserted {row_count} records into table `{table}`.")
+
+    def insert_new_records_to_production_for_section_and_user(
+        self, table: str, columns: List[str]
+    ) -> None:
+        """
+        Copies new records from the staging table to the production table. Specialized
+        for tables that have a foreign key to both LMSSection and LMSUser.
+
+        Parameters
+        ----------
+        table: str
+            Name of the table to truncate, not including the `stg_` prefix
+        columns: List[str]
+            A list of the column names in the table
+        """
+
+        assert table.strip() != "", "Argument `table` cannot be whitespace"
+        assert len(columns) > 0, "Argument `columns` cannot be empty"
+
+        insert_columns = ",".join(
+            [f"\n    {c}" for c in columns if not (c == "LMSSectionSourceSystemIdentifier" or c == "LMSUserSourceSystemIdentifier")]
+        )
+        select_columns = ",".join(
+            [
+                f"\n    stg.{c}"
+                for c in columns
+                if not (c == "LMSSectionSourceSystemIdentifier" or c == "LMSUserSourceSystemIdentifier")
```

Comment:
Only while doing the assignments upload did I realize how closely list comprehensions mirror SQL statements - when I had this kind of formatting.

Reply:
🤯